libraries/automaton/automaton.lua

```diff
@@ -0,0 +1,204 @@
+local items = require('items')
+local packet = require('packet')
+local player = require('player')
+local exp = require('expression')
+local resources = require('resources')
+local client = require('shared.client')
+local enumerable = require('enumerable')
+
+local command = require('core.command')
+
+local data, ftype = client.new('automaton_service')
+
+local res_items = resources.items
+local assembly_offset, attachment_offset = 0x2000, 0x2100
+
+ftype.fields.head = {
+    get = function(data)
+        return resources.items[data.head_id + assembly_offset]
+    end
+}
+
+ftype.fields.activated ={
+    get = function (data)
+        return player.pet_index ~= 0 --TODO: this does not allow swapping of attachment when your automaton is not the pet you have, which is wrong.
+                                     -- need to get the name of the pet and check if it matches automaton.name. getting the name seems best done a pet lib.
+    end
+}
+
+ftype.fields.frame = {
+     get = function(data)
+        return resources.items[data.frame_id + assembly_offset]
+    end
+}
+
+ftype.fields.attachments.type.base.fields.item = {
+    get = function(data)
+        return resources.items[data.id + attachment_offset]
+    end
+}
+
+local get_attachment_id = function(attachment)
+    local id = enumerable.wrap(items:find_ids(attachment)):first(exp.lookup(res_items):index('category'):is('Automaton'))
+    return id and id - attachment_offset or nil
+end
+
+local get_assembly_id = function(assembly)
```

Comment:
If you move the offset out of the function and to the call site you save a bit of code and you can merge the two functions.